soccer/gameplay/tactics/coordinated_pass.py

```diff
@@ -230,6 +241,17 @@ def on_enter_receiving(self):
         self.subbehavior_with_name('receiver').ball_kicked = True
         self.remove_subbehavior('kicker')
 
+    @property
+    def has_roles_assigned(self):
+        if self.state == CoordinatedPass.State.kicking:
+            return len(self.get_robots())==2
+        elif self.state == CoordinatedPass.State.receiving:
+            return len(self.get_robots())>=1
+        elif self.state == CoordinatedPass.State.preparing:
+            return True
+        else:
+            return False
```

Comment:
What's the context for this?

Reply:
I think this was used to detect failure in the restarts in an early attempt to stop the long runs on failure.